python/core/auto_generated/processing/qgsprocessingparameterdxflayers.sip.in

```diff
@@ -0,0 +1,112 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/core/processing/qgsprocessingparameterdxflayers.h                *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
+class QgsProcessingParameterDxfLayers : QgsProcessingParameterDefinition
+{
+%Docstring
+A parameter for Processing algorithms that need a list of input vector
+layers to export as DXF file - this parameter provides Processing
+framework's adapter for QList<QgsDxfExport.DxfLayer>.
+
+A valid value for this parameter is a list (QVariantList), where each
+item is a map (QVariantMap) in this form:
+{
+'layer':  string or QgsMapLayer,
+'attributeIndex': int
```

Comment:
One immediate thought -- I think this parameter should also be flexible enough to accept a list of plain layers (evaluated in the same way as a multi layer parameter). This would allow the parameter to be set in models from other algorithm outputs which output lists of layers only. (For similar reasons it should also accept a non-list, single layer)input type)

Reply:
In this case we will lost ability to configure split attribute, is this ok?

Reply:
I mean leave what you have, but make the evaluation tolerant and also accept layer lists and single layers as a fallback

Reply:
I got it. My point is that in this case we won't be able to tune split attribute. E.g. you can pass single layer, but can't tune how it should be exported.

Reply:
Should be fixed.

BTW, I'm wondering if the same requirement (with some sensible defaults) should be set for other algorithms e.g. mesh and vector tile writers? Right now, if I'm not wrong, we can't use them from modeler as they are not accepting list of plain layers (evaluated in the same way as a multi layer parameter) only their own custom inputs are allowed.

Reply:
Yep, that's a good point! They all should be forgiving and accept as many different input values as makes sense...